lib/auth/auth_with_roles_test.go

```diff
@@ -6759,6 +6759,138 @@ func TestGetActiveSessionTrackers(t *testing.T) {
 			},
 			checkSessionTrackers: require.Empty,
 		},
+		{
+			// an explicit deny on session_tracker still allows listing
+			// for sessions that the user can join
+			name: "explicit-deny-can-join",
+			makeRole: func() (types.Role, error) {
+				return types.NewRole("observe-sessions", types.RoleSpecV6{
+					Allow: types.RoleConditions{
+						JoinSessions: []*types.SessionJoinPolicy{
+							{
+								Name:  "observe-kube-sessions",
+								Kinds: []string{string(types.KubernetesSessionKind)},
+								Modes: []string{string(types.SessionObserverMode)},
+								Roles: []string{"access"},
+							},
+						},
+					},
+					Deny: types.RoleConditions{
+						Rules: []types.Rule{{
+							Resources: []string{types.KindSessionTracker},
+							Verbs:     []string{types.VerbList, types.VerbRead},
+						}},
+					},
+				})
+			},
+			extraSetup: func(t *testing.T, srv *TestTLSServer) {
+				originator, err := types.NewUser("session-originator")
+				require.NoError(t, err)
+
+				originator.AddRole("access")
+				originator, err = srv.Auth().UpsertUser(context.Background(), originator)
```

Comment:
Heads up, staticcheck error here because originator is written but not read (here and in the other tests)

```suggestion
				_, err = srv.Auth().UpsertUser(context.Background(), originator)
```